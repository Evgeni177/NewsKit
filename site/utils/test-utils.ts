import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
