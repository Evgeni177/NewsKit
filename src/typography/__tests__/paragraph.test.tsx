import {
  Paragraph,
  P,
  Subscript,
  Sub,
  Superscript,
  Sup,
  ParagraphProps,
} from '../paragraph';
import {renderToFragmentWithTheme} from '../../test/test-utils';

describe('Paragraph', () => {
  test('renders Paragraph', () => {
    const wrapper = renderToFragmentWithTheme(Paragraph, {
      children: 'paragraph component',
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('renders bold Paragraph', () => {
    const wrapper = renderToFragmentWithTheme(Paragraph, {
      children: 'paragraph component',
      $bold: true,
    } as ParagraphProps);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with drop cap', () => {
    const wrapper = renderToFragmentWithTheme(Paragraph, {
      children: 'paragraph component',
      $bold: true,
      $dropCap: true,
    } as ParagraphProps);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with coloured drop cap', () => {
    const wrapper = renderToFragmentWithTheme(Paragraph, {
      children: 'paragraph component',
      $bold: true,
      $dropCap: true,
      $dropCapColor: 'link010',
    } as ParagraphProps);
    expect(wrapper).toMatchSnapshot();
  });

  test('P alias for Paragraph', () => {
    expect(P).toEqual(Paragraph);
  });
});

describe('Subscript', () => {
  test('renders Subscript', () => {
    const wrapper = renderToFragmentWithTheme(Subscript, {
      children: 'subscript component',
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('Sub alias for Subscript', () => {
    expect(Sub).toEqual(Subscript);
  });
});

describe('Superscript', () => {
  test('renders Superscript', () => {
    const wrapper = renderToFragmentWithTheme(Superscript, {
      children: 'superscript component',
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('Sup alias for Superscript', () => {
    expect(Sup).toEqual(Superscript);
  });
});
