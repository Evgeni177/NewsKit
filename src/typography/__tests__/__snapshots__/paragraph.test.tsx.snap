// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders Paragraph 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 18px;
  line-height: 1.5555555555555556;
  font-weight: 400;
  -webkit-letter-spacing: -0.25px;
  -moz-letter-spacing: -0.25px;
  -ms-letter-spacing: -0.25px;
  letter-spacing: -0.25px;
  margin-bottom: 18px;
}

<p
    class="emotion-0"
  >
    paragraph component
  </p>
</DocumentFragment>
`;

exports[`Paragraph renders bold Paragraph 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 18px;
  line-height: 1.5555555555555556;
  font-weight: 400;
  -webkit-letter-spacing: -0.25px;
  -moz-letter-spacing: -0.25px;
  -ms-letter-spacing: -0.25px;
  letter-spacing: -0.25px;
  margin-bottom: 18px;
  font-weight: 500;
}

<p
    class="emotion-0"
  >
    paragraph component
  </p>
</DocumentFragment>
`;

exports[`Paragraph renders with coloured drop cap 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 18px;
  line-height: 1.5555555555555556;
  font-weight: 400;
  -webkit-letter-spacing: -0.25px;
  -moz-letter-spacing: -0.25px;
  -ms-letter-spacing: -0.25px;
  letter-spacing: -0.25px;
  margin-bottom: 18px;
  font-weight: 500;
}

.emotion-0:first-letter {
  float: left;
  padding-right: 0.15em;
  font-size: 5.34em;
  line-height: 0.85;
  color: #2F75D7;
}

<p
    class="emotion-0"
  >
    paragraph component
  </p>
</DocumentFragment>
`;

exports[`Paragraph renders with drop cap 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 18px;
  line-height: 1.5555555555555556;
  font-weight: 400;
  -webkit-letter-spacing: -0.25px;
  -moz-letter-spacing: -0.25px;
  -ms-letter-spacing: -0.25px;
  letter-spacing: -0.25px;
  margin-bottom: 18px;
  font-weight: 500;
}

.emotion-0:first-letter {
  float: left;
  padding-right: 0.15em;
  font-size: 5.34em;
  line-height: 0.85;
}

<p
    class="emotion-0"
  >
    paragraph component
  </p>
</DocumentFragment>
`;

exports[`Subscript renders Subscript 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 80%;
}

<sub
    class="emotion-0"
  >
    subscript component
  </sub>
</DocumentFragment>
`;

exports[`Superscript renders Superscript 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 80%;
}

<sup
    class="emotion-0"
  >
    superscript component
  </sup>
</DocumentFragment>
`;
